Automated Smoketest
Sha: 8be699572f19fb84bf711685bd3f100e737af0bd
Build: failed
Number of Error(s): 6
/mnt/disk1/home/vamosax/smoketest/PR-9395/HPCC-Platform/ecl/eclagent/eclagent.cpp:3376:9: error: \xe2\x80\x98unique_ptr\xe2\x80\x99 is not a member of \xe2\x80\x98std\xe2\x80\x99
/mnt/disk1/home/vamosax/smoketest/PR-9395/HPCC-Platform/ecl/eclagent/eclagent.cpp:3376:41: error: expected primary-expression before \xe2\x80\x98>\xe2\x80\x99 token
/mnt/disk1/home/vamosax/smoketest/PR-9395/HPCC-Platform/ecl/eclagent/eclagent.cpp:3376:43: error: \xe2\x80\x98serverstatus\xe2\x80\x99 was not declared in this scope
make[2]: *** [ecl/hthor/CMakeFiles/hthor.dir/__/eclagent/eclagent.cpp.o] Error 1
make[1]: *** [ecl/hthor/CMakeFiles/hthor.dir/all] Error 2
make: *** [all] Error 2

--- a/ecl/eclagent/eclagent.cpp
+++ b/ecl/eclagent/eclagent.cpp
@@ -52,6 +52,7 @@
 
 using roxiemem::OwnedRoxieString;
 
+#include <memory>
 #include <new>
 
 #ifdef _USE_CPPUNIT
@@ -3373,7 +3374,7 @@ extern int HTHOR_API eclagent_main(int argc, const char *argv[], StringBuffer * 
     try
     {
 #ifdef MONITOR_ECLAGENT_STATUS  
-        CSDSServerStatus * serverstatus = NULL;
+        std::unique_ptr<CSDSServerStatus> serverstatus;
 #endif
         Owned<ILocalWorkUnit> standAloneWorkUnit;
         if (wuXML)
@@ -3392,7 +3393,7 @@ extern int HTHOR_API eclagent_main(int argc, const char *argv[], StringBuffer * 
                 initClientProcess(serverGroup, DCR_EclAgent, 0, NULL, NULL, MP_WAIT_FOREVER);
             }
 #ifdef MONITOR_ECLAGENT_STATUS  
-            serverstatus = new CSDSServerStatus("ECLagent");
+            serverstatus.reset(new CSDSServerStatus("ECLagent"));
             serverstatus->queryProperties()->setPropInt("Pid", GetCurrentProcessId());
             serverstatus->commitProperties();
 #endif
@@ -3546,9 +3547,6 @@ extern int HTHOR_API eclagent_main(int argc, const char *argv[], StringBuffer * 
         catch (RELEASE_CATCH_ALL)
         {
         }
-
-        if (serverstatus)
-            delete serverstatus;
     }
     catch (IException * e)
     {

